### DISCUSSION TITLE
Adding `size=X` to `sns.scatterplot` adds an entry to the legend (as well as changing the pointsize)

### DISCUSSION THREAD

**Discussion by beyarkay** [mwaskom/seaborn#3490] [Q&A]
Minimum working example:
```py
import numpy as np
import seaborn as sns
import pandas as pd

data = pd.DataFrame()
data['x'] = np.random.random((100,))
data['y'] = np.random.random((100,))
data['hue'] = np.random.choice(['hue1', 'hue2', 'hue3'], (100,))
sns.scatterplot(
    data=data,
    x='x',
    y='y',
    hue='hue',
    size=1,
)
```

Result:

![Screenshot 2023-09-18 at 12 08 14](https://github.com/mwaskom/seaborn/assets/33420535/a36e8811-0d24-4f51-9724-fba6c7fff3a3)

I'm not entirely sure what's going on here, and it's possible that it's an error on my part. There seems to be many different types of behaviour depending on the usage of `hue`, `size`, and `s`:

<details>
<summary><code>sns.scatterplot(..., size=1, ...)</code> (item added to legend)</summary>

![size equals 1](https://github.com/mwaskom/seaborn/assets/33420535/b4c4f314-5e38-4f09-a283-972061540408)

</details>

<details>
<summary><code>sns.scatterplot(..., hue='hue', size=1, ...)</code> (item added to legend)</summary>

![Screenshot 2023-09-18 at 12 14 53](https://github.com/mwaskom/seaborn/assets/33420535/3402c1aa-a441-4317-a109-c03ce7c8e343)

</details>

<details>
<summary><code>sns.scatterplot(..., hue='hue', s=1, ...)</code> (no extra legend item)</summary>

![Screenshot 2023-09-18 at 12 15 16](https://github.com/mwaskom/seaborn/assets/33420535/c14680e8-853a-4d3e-af92-0f7c9a119fb8)

</details>

<details>
<summary><code>sns.scatterplot(..., hue='hue', ...)</code>(no extra legend item)</summary>

![Screenshot 2023-09-18 at 12 15 05](https://github.com/mwaskom/seaborn/assets/33420535/84310f2b-0486-43c6-b1b6-4420ded3849c)

</details>

I'd expect that adding `size=1` would not add an item `1` to the legend. Not sure if this is a bug or if I'm using the interface wrong though.

**Comment by mwaskom** [comment]
`size` is a mapping variable, like `hue`. When you pass a scalar, it gets broadcast, so it's equivalent to mapping with a vector of all 1s.

**Comment by beyarkay** [comment]
Okay thanks. I'm not sure why it gets added as an item in the legend though?

![](https://user-images.githubusercontent.com/33420535/268614142-a36e8811-0d24-4f51-9724-fba6c7fff3a3.png)

**Comment by mwaskom** [comment]
Why wouldn't it? The legend annotates the mapping variables, and you have a size mapping variable.
